File: 10.facade/facade.java
// Subsistemas
class AudioSystem {
    public void playAudio() {
        System.out.println("Reproduciendo audio...");
    }
}

class VideoSystem {
    public void playVideo() {
        System.out.println("Reproduciendo video...");
    }
}

// Facade
class MediaFacade {
    private AudioSystem audioSystem;
    private VideoSystem videoSystem;

    public MediaFacade() {
        audioSystem = new AudioSystem();
        videoSystem = new VideoSystem();
    }

    public void playMedia() {
        audioSystem.playAudio();
        videoSystem.playVideo();
    }
}

// Uso
public class facade {
    public static void main(String[] args) {
        MediaFacade mediaFacade = new MediaFacade();
        mediaFacade.playMedia();
    }
}

File: 10.facade/facade.ts
// Subsistemas
class AudioSystem {
    public playAudio(): void {
        console.log("Reproduciendo audio...");
    }
}

class VideoSystem {
    public playVideo(): void {
        console.log("Reproduciendo video...");
    }
}

// Facade
class MediaFacade {
    private audioSystem: AudioSystem;
    private videoSystem: VideoSystem;

    constructor() {
        this.audioSystem = new AudioSystem();
        this.videoSystem = new VideoSystem();
    }

    public playMedia(): void {
        this.audioSystem.playAudio();
        this.videoSystem.playVideo();
    }
}

// Uso
const mediaFacade = new MediaFacade();
mediaFacade.playMedia();